contracts/factory/src/contract.rs

```diff
@@ -537,10 +537,70 @@ impl Factory {
 
     #[allow(dead_code)]
     //TODO: Remove after we've added the key to storage
-    pub fn add_new_key_to_storage(env: Env) -> Result<(), ContractError> {
+    pub fn add_contract_name_key_to_storage(env: Env) -> Result<(), ContractError> {
         env.storage().persistent().set(&FACTORY_KEY, &true);
         Ok(())
     }
+
+    #[allow(dead_code)]
+    pub fn extend_all_ttl(env: Env) -> Result<(), ContractError> {
+        env.storage()
+            .instance()
+            .extend_ttl(INSTANCE_RENEWAL_THRESHOLD, INSTANCE_TARGET_TTL);
+
+        // extend all datakeys
+        for key in &[DataKey::Config, DataKey::LpVec, DataKey::Initialized] {
+            if env.storage().persistent().has(key) {
+                env.storage().persistent().extend_ttl(
+                    key,
+                    PERSISTENT_RENEWAL_THRESHOLD,
+                    PERSISTENT_TARGET_TTL,
+                );
+            } else {
+                return Err(ContractError::KeyNotFound);
+            }
+        }
```

Comment:
That error seems superflous since we know those keys are there. They are initiated during instantiation.